provider/server/exec.go

```diff
@@ -26,13 +24,16 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"k8s.io/client-go/tools/remotecommand"
+	"libvirt.org/go/libvirtxml"
 )
 
 const (
 	StreamCreationTimeout = 30 * time.Second
 	StreamIdleTimeout     = 2 * time.Minute
 )
 
+var activeConsoles sync.Map
```

Comment:
+1

Reply:
As discussed with @lukas016 offline, it gets initialized everytime (https://github.com/ironcore-dev/libvirt-provider/blob/531b8fe9089a807c8075a523ab77c7ed55720d88/provider/server/exec.go#L86). Hence keeping it in that struct won't serve the purpose.

Reply:
@lukasfrank we discuss it offline and we are not very happy with current solution where streaming server internally calling iri server method. I think it has to be improve with refactor so i agree with current solution.
You can put it into Server structure but a lot of functionality are methods of Server so you won't have global variable but you will have variable accessible in all methods from Server object, what is for me extremely similar solution but i agree it will follow best practices.

Reply:
Since we anyhow have the `execRequestCache request.Cache[*iri.ExecRequest]` at the server which is used here, I would propose to attach the map to the server as well.

Reply:
Done